src/Microsoft.Extensions.Configuration.AzureAppConfiguration/Extensions/BytesExtensions.cs

```diff
@@ -0,0 +1,46 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+//
+using System.Text;
+using System;
+
+namespace Microsoft.Extensions.Configuration.AzureAppConfiguration.Extensions
+{
+    internal static class BytesExtensions
+    {
+        /// <summary>
+        /// Converts a byte array to Base64URL string with optional padding ('=') characters removed.
+        /// Base64 description: https://datatracker.ietf.org/doc/html/rfc4648.html#section-4
+        /// </summary>
+        public static string ToBase64Url(this byte[] bytes)
+        {
+            string bytesBase64 = Convert.ToBase64String(bytes);
+
+            int indexOfEquals = bytesBase64.IndexOf("=");
+
+            // Remove all instances of "=" at the end of the string that were added as padding
+            int stringBuilderCapacity = indexOfEquals != -1 ? indexOfEquals : bytesBase64.Length;
+
+            StringBuilder stringBuilder = new StringBuilder(stringBuilderCapacity);
+
+            // Construct Base64URL string by replacing characters in Base64 conversion that are not URL safe
```

Comment:
Is there a reason the featureFlagId needs to be URL safe?

Reply:
For use in REST resource IDs (/experiment/{id})

Reply:
So, sounds like we should have a comment in CalculateFeatureFlagId to explain why we need base64url encoding.

Reply:
That seems more like a design doc/specification territory rather than a code comment. The code just implements the spec of a feature flag id, which uses b64url encoding.